packages/hub/utils/scheduled-payments.ts

```diff
@@ -14,6 +15,13 @@ export function calculateNextPayAt(fromDate: Date, recurringDay: number) {
   let adjustedRecurringDay = Math.min(recurringDay, daysInMonth(fromDate));
   if (fromDate.getUTCDate() < adjustedRecurringDay) {
     nextPayAtUtc.setUTCDate(adjustedRecurringDay); // next pay this month
+  } else if (recurringUntil && fromDate > recurringUntil && fromDate.getUTCDate() === adjustedRecurringDay) {
```

Comment:
Instead, would it be simpler to adjust this function like so?

Instead of:

```
return nextPayAtUtc;
```

We can do this:

```
if (nextPayAtUtc <= recurringUntilUtc) {
  return nextPayAtUtc;
} else {
  return null; // Payment expired, there is no next pay date
}

Reply:
Hmm, are we allowing a user to schedule recurring payments with recurring until before `dateNow()`?

Reply:
We should not allow that.